pandas/sparse/array.py

```diff
@@ -769,14 +769,17 @@ def make_sparse(arr, kind='block', fill_value=None):
     if isnull(fill_value):
         mask = notnull(arr)
     else:
-        mask = arr != fill_value
+        mask = np.not_equal(arr, fill_value)
+        # In NumPy 1.12.0, not implemented at least for arr.dtype=str
+        if mask is NotImplemented:
+            mask = np.not_equal(arr.astype(object), fill_value)
```

Comment:
I needed to change this bit in order to support adding a `str` (`object`) column to a `SparseDataFrame` with `default_fill_value=0`. See here: https://github.com/pandas-dev/pandas/pull/15497/files#diff-b30cded911d05ab8f4b2ab86c7dcab58R1169
Otherwise it would crash with some uninformative and unexpected error.